cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ExecutorStateProcessor.java

```diff
@@ -0,0 +1,115 @@
+        return executorState.get(FINISHED_DATA_MOVEMENT_KEY).asInt();
+    }
+
+    /**
+     * Extracts the task start time from the provided `executorState` JSON object.
+     * The task start time is extracted from the "triggeredTaskReason" field, which contains a
+     * timestamp in ISO 8601 format. The timestamp is then parsed into seconds since the Unix epoch.
+     *
+     * @param executorState The `JsonNode` object containing the state of the executor,
+     *                      from which the task start time will be extracted.
+     * @return The task start time in seconds since the Unix epoch.
+     */
+    public static Integer getTaskStartTime(JsonNode executorState) {
+        if (!executorState.has(TRIGGERED_TASK_REASON_KEY)) {
+            throw new IllegalArgumentException(String.format("Executor State does not contain required '%s' field.", TRIGGERED_TASK_REASON_KEY));
+        }
+        String triggeredTaskReason = executorState.get(TRIGGERED_TASK_REASON_KEY).asText();
+        // Extract the timestamp from the string, assuming it's in ISO 8601 format
+        String dateString = extractDateFromTriggeredTaskReason(triggeredTaskReason);
+        return (int) parseDateToSeconds(dateString);
+    }
+
+    /**
+     * Method to parse date-time string in ISO 8601 into time in seconds since Unix epoch.
+     *
+     * @param dateString Date-time string in ISO 8601 format.
+     *
+     * @return Seconds since Unix epoch.
+     */
+    private static long parseDateToSeconds(String dateString) {
+        // Validate the date format
+        if (dateString == null || dateString.isEmpty()) {
+            throw new IllegalArgumentException("Invalid date string.");
+        }
+        // Use Instant to parse the date-time string in ISO 8601 format
+        Instant instant = Instant.parse(dateString);
+        return instant.getEpochSecond();  // Convert to seconds
+    }
+
+    /**
+     * Extracts the ISO 8601 date-time string from a Cruise Control task's reason string.
+     * The date-time is expected to be in the format "%s (Client: %s, Date: %s)", where
+     * the date follows "Date:" in UTC and is rounded to the second
+     * (see: https://github.com/linkedin/cruise-control/blob/main/cruise-control-core/src/main/java/com/linkedin/cruisecontrol/CruiseControlUtils.java#L39-L41).
+     *
+     * @param triggeredTaskReason Cruise Control task's triggeredTaskReason string.
+     *
+     * @return Date-time string in ISO 8601 format.
+     */
+    private static String extractDateFromTriggeredTaskReason(String triggeredTaskReason) {
+        if (triggeredTaskReason == null || triggeredTaskReason.isEmpty()) {
+            throw new IllegalArgumentException("Triggered task reason is missing.");
+        }
+
+        // Regular expression pattern to match the date-time in ISO 8601 format (UTC, rounded to the second).
+        String regex = "\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}Z";
+        Pattern pattern = Pattern.compile(regex);
```

Comment:
I would compile the pattern as a static private field of the class, otherwise this compile is executed every time we parse a date so losing the advantage (for performance) of having the pattern compiled and ready already.

Reply:
Fixed.